tbx/project_styleguide/templates/patterns/navigation/components/primary-nav.html

```diff
@@ -1,20 +1,20 @@
 {% load wagtailcore_tags %}
 
 {% for item in primarynav %}
-    {% with children=item.value.page.get_children.live.public.in_menu hide_children=item.value.hide_children %}
-        <li class="primary-nav-desktop__item {% if children and not hide_children %}primary-nav-desktop__item--is-parent{% endif %}"
-            {% if children and not hide_children %}data-has-subnav{% endif %}
+    {% with children=item.value.page.get_children.live.public.in_menu child_display=item.value.child_display_behaviour %}
+        <li class="primary-nav-desktop__item {% if children and child_display != 'hide_children' %}primary-nav-desktop__item--is-parent{% endif %}"
+            {% if children and not child_display %}data-has-subnav{% endif %}
```

Comment:
Thanks fo spotting that!

I was also wondering if there could be a way to improve this. 

Perhaps defining custom methods via the [`StructValue` approach](https://docs.wagtail.org/en/v5.2.3/advanced_topics/customisation/streamfield_blocks.html#additional-methods-and-properties-on-structblock-values):

```python
class PrimaryNavLinkStructValue(blocks.StructValue):
    def hide_children(self):
        # return True if child_display_behaviour == ChildDisplay.HIDE 

    def show_up_to_level1(self):
        # return True if child_display_behaviour == ChildDisplay.SHOW_UP_TO_LEVEL1
        
    def show_up_to_level2(self):
        # return True if child_display_behaviour == ChildDisplay.SHOW_UP_TO_LEVEL2
```

Then, instead of

https://github.com/torchbox/torchbox.com/blob/2d0de5811e95aa16dbc1cddb2f634f2545ddf867/tbx/project_styleguide/templates/patterns/navigation/components/primary-nav.html#L4

we could do

```django-html
{% with children=item.value.page.get_children.live.public.in_menu hide_children=item.value.hide_children show_up_to_level1=item.value.show_up_to_level1 show_up_to_level2=item.value.show_up_to_level2 %}
```

Reply:
as discussed this morning let's leave this now and I'll just use some clear commenting in the template to try and make it really clear.

Reply:
Fixed the bug in 01441b2